### PR TITLE
[TOOL-165]step4. Add the Migration Lambda in with it's current code

### PR DESCRIPTION
**Pull request by paperx123**
Change
- move original code with work-around in source code at this point
   -- use test webhook url 
   -- log at the begining of the handler
- workaround ts config at this point
   -- turn off check upon 'any'   
- update in serverless.yml to add the new endpoint
    -- add iam 
    -- add function definition
- install required packages

Test
- sls deployment 
- migration-staging trigger
- checkout the log 
Screen Shot 2019-11-29 at 10 29 19 AM
Screen Shot 2019-11-29 at 10 44 33 AM
Screen Shot 2019-11-29 at 10 44 58 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,9 @@
     "url": "https://github.com/conqa/migration-slack-logger/issues"
   },
   "homepage": "https://github.com/conqa/migration-slack-logger#readme",
-  "dependencies": {},
+  "dependencies": {
+    "aws-sdk": "^2.580.0"
```
Comment (wrumsby) [inline package.json]: This doesn't need to be a dependency as it is [provided by the lambda runtime](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html).

In other projects we are specifying this in `devDependencies`.

I'd run:

```
npm uninstall aws-sdk
npm install -D --save-exact aws-sdk@2.536.0
```

The reason I'd use `--save-exact` here is that we want to use exactly what the runtime is providing (if we're going to rely on the runtime providing this).

File: serverless.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ provider:
   region: ${opt:region, "us-west-2"}
   iamManagedPolicies:
     - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+    - arn:aws:iam::aws:policy/service-role/AWSLambdaDynamoDBExecutionRole
```
Comment (wrumsby) [inline serverless.yml]: I'd suggest you want a more specific role - i.e. one that targets the resources you specifically want to work with.

Recommend looking at [servleress-iam-roles-per-function](https://www.npmjs.com/package/serverless-iam-roles-per-function) and using that to specify more precise permissions.

File: serverless.yml [hunk 3]
```diff
@@ -33,3 +34,9 @@ functions:
           path: hello
     environment:
       CONQA_ENV: ${self:provider.stage}
+  bot:
+    handler: ${opt:buildDir}/bot.handler
+    events:
+      - stream: arn:aws:dynamodb:us-west-2:117276354576:table/migration-staging/stream/2019-11-28T05:32:08.616
```
Comment (wrumsby) [inline serverless.yml]: The ARN here looks like it is only for `staging` - so it's possibly the wrong resource (and you might want to consider having different resources based on `${self.provider.stage}`).

File: serverless.yml [hunk 4]
```diff
@@ -33,3 +34,9 @@ functions:
...
+    environment:
+      CONQA_ENV: ${self:provider.stage}
```
Comment (wrumsby) [inline serverless.yml]: Add `NODE_ENV: production` here too.

File: tsconfig.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "types": ["jest", "node"],
     "skipLibCheck": true,
     "esModuleInterop": true,
+    "noImplicitAny": false,
```
Comment (wrumsby) [inline tsconfig.json]: 😢

File: package.json (hunk 1 above)
Comment (paperx123) [reply package.json]: fixed

File: serverless.yml (hunk 2 above)
Comment (paperx123) [reply serverless.yml]: planned to deal with it in next PR. but updated with recent update on this same PR anyway

File: serverless.yml (hunk 4 above)
Comment (paperx123) [reply serverless.yml]: fixed

File: tsconfig.json (hunk 5 above)
Comment (paperx123) [reply tsconfig.json]: :-)  in next steps as in the ticket https://conqanz.atlassian.net/browse/TOOL-165?atlOrigin=eyJpIjoiMmEwNDYzMjhkZWEwNDc4Mjg3MmRiZTE5Y2Y2YzhiMGMiLCJwIjoiaiJ9

1.  PR to Put Node 12 onto template
2. Enable Server-less Support on Rep with example function with minimal IAM policy
3. Test a node 12 deploy of example function *Discuss with walter what to push back into Template Repo*
4. Add the Migration Lambda in with it's current code
5. Convert to Type Script & Fix warnings
6. Add Tests?
7. CI / CD
8. Delete The Old One

File: serverless.yml (hunk 3 above)
Comment (paperx123) [reply serverless.yml]: here is for testing only at this stage. it'll be changed afterwards

Comment (wrumsby) [reply serverless.yml]: Fair enough - maybe a `TODO:` or `FIXME:` comment might help so this doesn't get forgotten?

File: serverless.yml [hunk 6]
```diff
@@ -33,3 +42,21 @@ functions:
           path: hello
     environment:
       CONQA_ENV: ${self:provider.stage}
+      NODE_ENV: production
+  bot:
+    handler: ${opt:buildDir}/bot.handler
+    iamManagedPolicies:
+      - arn:aws:iam::aws:policy/service-role/AWSLambdaDynamoDBExecutionRole
```
Comment (wrumsby) [inline serverless.yml]: I'm pretty sure that since you have the specific inline policy you don't need this broader policy.

**Pull request merged**